docs/shared/recipes/generators/local-generators.md

```diff
@@ -117,3 +117,49 @@ Nx uses the paths from `tsconfig.base.json` when running plugins locally, but us
 ## Generator Utilities
 
 The [`@nx/devkit` package](/nx-api/devkit/documents/nx_devkit) provides many utility functions that can be used in generators to help with modifying files, reading and updating configuration files, and working with an Abstract Syntax Tree (AST).
+
+## Unit Testing Generators
+
+Generators will be created with a default test suite. You can add additional tests to this test suite to ensure that your generator is working as expected.
+
+If your generator makes use of or composes generators that make use of Inferred Tasks or Inference Plugins, you will need additional setup in your test suite to ensure that any plugins registered in your current Nx Workspace do not interfere with the unit tests.
```

Comment:
I don't like this. :confused: 

This is indeed the case.. but how are developers supposed to know that generators they're importing are using inferences plugins or not?

Can we explore creating a `TempFs` when we create a tree? :eyes: Which is already public testing API.. and used.

Reply:
We don’t currently require tear down logic for when unit testing with create tree, because it’s all in-memory.

To change that API to start creating TempFs would mean countless unit tests in countless projects would start creating real temp dirs on real machines, that would not be cleaned up.

As for discoverability of whether the generator uses inference or not, that’s a separate issue.
But they should at least have an escape hatch right now if they run into it.

currently they have nothing

Reply:
We could create an entirely new `createTreeWithTempFs` utility that could be used with the caveat that tear down logic _should_ be run to prevent build up of temp dirs

Reply:
I have an idea :thinking:

Reply:
I'm going to close this PR as not the correct way forward - but I'm glad the service in this was able to help :)